### DISCUSSION TITLE
Get site by ID

### DISCUSSION THREAD

**Discussion by wonderplayer** [pnp/pnpcore#398] [Q&A] [answered]
Is there a way to get site by ID?
My scenario is that I have hub site ID and I would like to get that hub site using that ID.
I know that there is REST API for that, and actually found out that there are tests, where this is used: [test class](https://github.com/pkbullock/pnpcore/blob/4ef9ee67a86d14e320b821292e22a0647fb4eec3/src/sdk/PnP.Core.Test/SharePoint/HubSiteTests.cs#L61) and [HubSite class](https://github.com/pkbullock/pnpcore/blob/4ef9ee67a86d14e320b821292e22a0647fb4eec3/src/sdk/PnP.Core/Model/SharePoint/Settings/Internal/HubSite.cs#L10).
I was wondering if it is possible to achieve the same in any way using SDK without calling REST API directly?

Thank you!

**Comment by jansenbe** [comment, marked as answer]
@wonderplayer : good idea, some additional work is needed to make this easier. I'll add an issue for it to track the need

**Reply by jansenbe** [reply]
Hi @wonderplayer ,

In the meantime @pkbullock added support to get hub site information from a hub site id as explained here: https://pnp.github.io/pnpcore/using-the-sdk/sites-hubsites.html#get-hub-site-data. Hope that's sufficient to realize your scenario?

**Reply by wonderplayer** [reply]
Yes, it will be sufficient for my scenario 😊
I already implemented logic to use REST API directly, but this will definitely simplify stuff, thank you!

**Reply by pkbullock** [reply]
Any issues let us know, thank you for the feedback!
